src/java/org/apache/cassandra/index/sai/disk/format/IndexComponent.java

```diff
@@ -79,10 +84,20 @@ public enum IndexComponent
      */
     GROUP_COMPLETION_MARKER("GroupComplete");
 
+    public final Component.Type type;
     public final String name;
 
     IndexComponent(String name)
     {
         this.name = name;
+        this.type = componentType(name);
+    }
+
+    private static Component.Type componentType(String name)
+    {
+        return Component.Type.create(SAI_DESCRIPTOR + SAI_SEPARATOR + name.toUpperCase(),
```

Comment:
nit: `toUpperCase()` here seems unnecessary, as it will give us weird things like "GROUPCOMPLETE". Camel is ok?

Reply:
Indeed. Fixed